### PR TITLE
feat: Command Sorting

### PR DESCRIPTION
**Pull request by TAHuntling**
feat: Command Organization

Description: Changed CommandInfos to be a map rather than a vector, which allows for easy alphabetization of the different commands available when running /help.

Motivation and Context: Found it hard to sort through the list of commands as it was, so this should be helpful for users to find the commands they are looking for.

How Has This Been Tested?: Running my server and using the help command, for it to show the resulting command list in alphabetical order

### CODE REVIEW

**Comment by EmosewaMC** [comment]
The files are still not formatted to our standard.  Please format them

**Comment by EmosewaMC** [comment]
All feedback from the other pr aside from the lambda to function still must be addressed

**Comment by TAHuntling** [comment]
> The files are still not formatted to our standard. Please format them

Is it formatted to the standard now?

**Comment by EmosewaMC** [comment]
> > The files are still not formatted to our standard. Please format them
> 
> Is it formatted to the standard now?

it is not.  there have been no pushes to this branch yet since my comment on the last pr

**Comment by TAHuntling** [comment]
> > > The files are still not formatted to our standard. Please format them
> > 
> > 
> > Is it formatted to the standard now?
> 
> it is not. there have been no pushes to this branch yet since my comment on the last pr

how abour now

**Comment by EmosewaMC** [comment]
> > > > The files are still not formatted to our standard. Please format them
> > > 
> > > 
> > > Is it formatted to the standard now?
> > 
> > 
> > it is not. there have been no pushes to this branch yet since my comment on the last pr
> 
> how abour now

It is not formatted correctly after that push, if anything that push just made more lines 4 spaces instead of a tab

**Comment by TAHuntling** [comment]
> It is not formatted correctly after that push, if anything that push just made more lines 4 spaces instead of a tab

Ok I set up the tab stops setting and tried it, did it work?

**Comment by EmosewaMC** [comment]
> > It is not formatted correctly after that push, if anything that push just made more lines 4 spaces instead of a tab
> 
> Ok I set up the tab stops setting and tried it, did it work?

you didnt push your changes

File: dGame/dUtilities/SlashCommandHandler.cpp [hunk 1]
```diff
@@ -74,41 +71,37 @@ void SlashCommandHandler::HandleChatCommand(const std::u16string& chat, Entity*
...
+				feedback << "\nAliases: ";
+				for (size_t i = 0; i < it->second.aliases.size(); i++) {
+					if (i > 0) feedback << ", ";
+					feedback << it->second.aliases[i];
+				}
 			}
+		} else {
+			// Let GameMasters know if the command doesn't exist or they don't have access
+			if (entity->GetGMLevel() > eGameMasterLevel::CIVILIAN) {
+				feedback << "Command " << std::quoted(args) << "Does Not Exist";
```
Comment (EmosewaMC) [inline dGame/dUtilities/SlashCommandHandler.cpp:99]:
```suggestion
				feedback << "Command " << std::quoted(args) << "does not exist";
```

File: dGame/dUtilities/SlashCommandHandler.cpp [hunk 2]
```diff
@@ -37,10 +35,11 @@ void SlashCommandHandler::RegisterCommand(Command command) {
 			LOG_DEBUG("Command alias %s is already registered! Skipping!", alias.c_str());
 			continue;
 		}
+
+		CommandInfos.insert(std::make_pair(command.aliases[0], command));
```
Comment (EmosewaMC) [inline dGame/dUtilities/SlashCommandHandler.cpp:39]:
```suggestion
		CommandInfos[command.aliases[0]] = command;
```

File: dGame/dUtilities/SlashCommandHandler.cpp [hunk 3]
```diff
@@ -19,16 +19,14 @@
...
+	std::map<std::string, Command> CommandInfos;
 	std::map<std::string, Command> RegisteredCommands;
 }
 
 void SlashCommandHandler::RegisterCommand(Command command) {
-	if (command.aliases.empty()) {
-		LOG("Command %s has no aliases! Skipping!", command.help.c_str());
-		return;
+	if (!command.aliases.empty()) {
+		CommandInfos[command.aliases[0]] = command;
```
Comment (EmosewaMC) [inline dGame/dUtilities/SlashCommandHandler.cpp:28]:
```suggestion
		LOG("Command %s has no aliases! Skipping!", command.help.c_str());
		return;
```

File: dGame/dUtilities/SlashCommandHandler.cpp [hunk 4]
```diff
@@ -19,16 +19,14 @@
...
-	std::vector<Command> CommandInfos;
+	std::map<std::string, Command> CommandInfos;
 	std::map<std::string, Command> RegisteredCommands;
 }
 
 void SlashCommandHandler::RegisterCommand(Command command) {
-	if (command.aliases.empty()) {
-		LOG("Command %s has no aliases! Skipping!", command.help.c_str());
-		return;
+	if (!command.aliases.empty()) {
```
Comment (EmosewaMC) [inline dGame/dUtilities/SlashCommandHandler.cpp:27]:
```suggestion
	if (command.aliases.empty()) {
```

File: dGame/dUtilities/SlashCommandHandler.cpp [hunk 5]
```diff
@@ -74,41 +73,36 @@ void SlashCommandHandler::HandleChatCommand(const std::u16string& chat, Entity*
...
+			feedback << "----- " << args << " -----\n";
+			feedback << it->second.info;
+			if (it->second.aliases.size() > 1) {
+				feedback << "\nAliases: ";
+				for (size_t i = 0; i < it->second.aliases.size(); i++) {
+					if (i > 0) feedback << ", ";
+					feedback << it->second.aliases[i];
+				}
 			}
+		} else {
```
Comment (EmosewaMC) [inline dGame/dUtilities/SlashCommandHandler.cpp:98]:
```suggestion
		} else if (entity->GetGMLevel() > eGameMasterLevel::CIVILIAN) {
```

File: dGame/dUtilities/SlashCommandHandler.cpp [hunk 6]
```diff
@@ -74,41 +73,36 @@ void SlashCommandHandler::HandleChatCommand(const std::u16string& chat, Entity*
...
+			if (entity->GetGMLevel() > eGameMasterLevel::CIVILIAN) {
```
Comment (EmosewaMC) [inline dGame/dUtilities/SlashCommandHandler.cpp:99]:
```suggestion
```

File: dGame/dUtilities/SlashCommandHandler.cpp [hunk 7]
```diff
@@ -74,41 +73,36 @@ void SlashCommandHandler::HandleChatCommand(const std::u16string& chat, Entity*
...
+				feedback << "Command " << std::quoted(args) << "does not exist";
+			};
```
Comment (EmosewaMC) [inline dGame/dUtilities/SlashCommandHandler.cpp:101]:
```suggestion
```

File: dGame/dUtilities/SlashCommandHandler.cpp [hunk 8]
```diff
@@ -38,9 +38,40 @@ void SlashCommandHandler::RegisterCommand(Command command) {
 			continue;
 		}
 	}
+	CommandInfos.insert(std::make_pair(command.aliases[0], command));
```
Comment (EmosewaMC) [inline dGame/dUtilities/SlashCommandHandler.cpp:41]:
```suggestion
	CommandInfos[command.aliases[0]] = command;
```

**Comment by TAHuntling** [comment]
> Pending CI. Please also post the tests you did on the last commit with changes to this branch

https://youtu.be/vzjk0NxxUus

**Review by aronwk-aaron — APPROVED** [review]

**Pull request merged**